blocks/bricks/sequence_generators.py

```diff
@@ -136,6 +137,34 @@ def _push_allocation_config(self):
 
     @application
     def cost(self, application_call, outputs, mask=None, **kwargs):
+        """Returns sum of generation costs for output sequences.
+
+        Parameters
+        ----------
+        outputs : :class:`~tensor.TensorVariable`
+            The 3(2) dimensional tensor containing output sequences.
+            The dimension 0 must stand for time, the dimension 1 for the
+            position on the batch.
+        mask : :class:`~tensor.TensorVariable`
+            The binary matrix identifying fake outputs.
+
+        Notes
+        -----
+        The contexts are expected as keyword arguments.
+
+        """
+        # Compute the sum of costs
+        cost = self.cost_matrix(outputs, mask=mask, **kwargs).sum()
```

Comment:
But if you do not normalize at all, then you have to decrease the learning rate when you increase the batch size. 

At least it has to be done uniformly for all `COST` variables in Blocks. `CategoricalCrossEntropy` and `CostMatrix` normalize costs.